draft-birkholz-scitt-architecture.md

```diff
@@ -552,64 +575,126 @@ Editor's Note: This may be moved to appendix.
+One of the following:
 
-`InvalidInput` means either the Signed Claims message is syntactically malformed, violates the signing profile (e.g. signing algorithm), or has an invalid signature relative to the currently resolved DID document.
+- Status 200 - Registration is pending
+    - Header: `Content-Type: application/json`
+    - (Optional) Header: `Retry-After: <seconds>`
+    - Body: `{ "operationId": "<Operation ID>", "status": "pending" }`
 
-### Retrieve Registration Receipt
+- Status 200 - Registration was successful
+    - Header: `Location: <Base URL>/entries/<Entry ID>`
+    - Header: `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "registered", "entryId": "<Entry ID>" }`
+
+- Status 200 - Registration failed
+    - Header `Content-Type: application/json`
+    - Body: `{ "operationId": "<Operation ID>", "status": "failed", "error": { "code": "<code>", "message": "<message>" } }`
+    - Error code: `InvalidInput`
+
+- Status 404 - Unknown Operation ID
+    - Error code: `NotFound`
+    - This can happen if the operation ID has expired and been deleted.
+
+If an operation ID is invalid (i.e., it does not correspond to any submit operation), a service may return either a 404 or a `pending` status. This is because differentiating between the two may not be possible in an eventually consistent system.
+
+### Retrieve Registration Entry
 
 #### Request
 
 ~~~
-GET <Base URL>/entries/<Transaction ID>/receipt
+GET <Base URL>/entries/<Entry ID>
 ~~~
 
 #### Response
 
 One of the following:
 
-- HTTP Status `200` - Registration was successful and the Receipt is returned.
-- HTTP Status `400` - Transaction exists but does not correspond to a Registration Request.
-  - Error code `TransactionMismatch`
-- HTTP Status `404` - Transaction is pending, unknown, or invalid.
-  - Error code `TransactionPendingOrUnknown`
-  - Error code `TransactionInvalid`
+- Status 200.
+  - Header `Content-Type: application/json`
+  - Body `{ "entryId": "<Entry ID>", "status": "registered" }`
+- Status 404 - Entry not found.
+  - Error code: `NotFound`
+
+### Retrieve Registered Claim
 
-The `200` response contains the SCITT_Receipt in the body.
+#### Request
 
-The `400` and `404` responses return the error details as described earlier.
+~~~
+GET <Base URL>/entries/<Entry ID>/claim
+~~~
 
-The retrieved Receipt may be embedded in the corresponding COSE_Sign1 document in the unprotected header, see TBD.
+Query parameters:
+
+- (Optional) `embedReceipt=true`
+
+If the query parameter `embedReceipt=true` is provided, then the claim is returned with the corresponding registration receipt embedded in the claim's COSE unprotected header.
+
+#### Response
+
+One of the following:
+
+- Status 200.
+  - Header: `Content-Type: application/cose`
```

Comment:
STD96 defines `application/cose; cose-type="cose-sign1"` for this

Reply:
We required tagged COSE, so I'd rather avoid repeating it here for the sake of simpler clients.

Reply:
sure, but these are different layers.

unless you are only ever allowing COSE Sign1 in the body, in which case the type qualifier is redundant, this approach is prone to MIME sniffing, which in the long term makes clients more complex rather than simpler.

Reply:
My main worry with using a media type with parameter is that it's not trivial to match it. Some clients will add a whitespace, some won't, and the server has to decode the media type and parameters, instead of doing a simple string match.

Reply:
If we can avoid normative requirements for media type parameters, we should... not sure if thats possible in this case.

Reply:
@letmaik Are you sure that we will stick with "COSE Sign1 only" forever?

Reply:
@henkbirkholz No, but I think it'll be more likely that we use more concrete media types in the future, rather than the generic `application/cose;type=..`, like `application/scitt-claim+cose`. So, I'd prefer to treat this (`application/cose`) like for receipts (`application/cbor`) a valid placeholder.

Reply:
> @henkbirkholz No, but I think it'll be more likely that we use more concrete media types in the future, rather than the generic `application/cose;type=..`, like `application/scitt-claim+cose`. So, I'd prefer to treat this (`application/cose`) like for receipts (`application/cbor`) a valid placeholder.

If the plan is to move to more precise typing info (i.e., `application/scitt-claim+cose`) in the near future, then I'm happy.